### PR TITLE
[RFC] Fixes required to build P9 STOP API with skiboot

### PR DESCRIPTION
**Pull request by Over-enthusiastic**
This pull request contains code changes which are needed to build P9 STOP API for skiboot/opal firmware. 

- Fixes  are mostly around detecting skiboot environment using  \_\_SKIBOOT\_\_ and including files required for the build.

- Fixes related to using C compiler instead of C++

- Fixes related to mixed code and declaration and other compiler error/warnings.

---
This change is [Reviewable](https://reviewable.io/reviews/open-power/hostboot/100)

### CODE REVIEW

**Comment by dcrowell77** [comment]
Pulling in locally to get feedback from PM team that owns these files and PHYP team that uses them.

**Comment by dcrowell77** [comment]
Are you pulling this code out of the hostboot repo to compile into skiboot?  Or are these changes just there to keep things common with the version inside skiboot?

**Comment by Over-enthusiastic** [comment]
I am not pulling this code directly from the hostboot repo.
These changes are to keep things common with the version inside skiboot.

**Comment by stewart-ibm** [comment]
Akshay Adiga <notifications@github.com> writes:
> I am not pulling this code directly from the hostboot repo.
> These changes are to keep things common with the version inside skiboot.

This is what we ended up doing for the libstb and libpore parts on P8
(and will continue to to for libstb for P9).

Keeping things in sync is currently a bit of a manual affair though.


-- 
Stewart Smith
OPAL Architect, IBM.

File: src/import/chips/p9/procedures/utils/stopreg/p9_stop_api.C [hunk 1]
```diff
@@ -865,17 +874,11 @@ StopReturnCode_t p9_stop_save_scom( void* const   i_pImage,
...
-        ScomEntry_t* pEntryLocation = NULL;
-        ScomEntry_t* pNopLocation = NULL;
-        ScomEntry_t* pTableEndLocationtable = NULL;
-        uint32_t swizzleAddr = SWIZZLE_4_BYTE(i_scomAddress);
-        uint64_t swizzleData = SWIZZLE_8_BYTE(i_scomData);
-        uint32_t swizzleAttn = SWIZZLE_4_BYTE(ATTN_OPCODE);
-        uint32_t swizzleEntry = SWIZZLE_4_BYTE(SCOM_ENTRY_START);
-        uint32_t index = 0;
-        uint32_t swizzleBlr = SWIZZLE_4_BYTE(BLR_INST);
+	nopInst = getOriInstruction( 0, 0, 0 );
```
Comment (dcrowell77) [inline src/import/chips/p9/procedures/utils/stopreg/p9_stop_api.C]: From Vance - If you want identical behavior, I think you still want to NULL your 3 ScomEntry_t pointers here. There's a possibility a pointer gets set one time around the loop and not changed the next time around.

Comment (Over-enthusiastic) [inline src/import/chips/p9/procedures/utils/stopreg/p9_stop_api.C]: Oh yeh, I completely missed it.

**Pull request closed without merging**